frontend/context/BalanceProvider.tsx

```diff
@@ -114,7 +114,7 @@ export const BalanceProvider = ({ children }: PropsWithChildren) => {
   }, [selectedAgentConfig.evmHomeChainId, stakedBalances]);
 
   const updateBalances = useCallback(async () => {
-    if (masterWallets && services) {
+    if (!isNil(masterWallets) && !isEmpty(masterWallets) && !isNil(services)) {
```

Comment:
added this check because the balances were requested even when the account was not created yet